### ISSUE TITLE
The client cannot tells whether audience restriction has been applied

### ISSUE THREAD

**Issue by randomstuff** [oauth-wg/draft-ietf-oauth-resource-metadata#55]
The current draft contains the wording:

> While this attack is not explicitly enabled by this specification, and is possible in a plain OAuth 2.0 deployment, it is made somewhat more likely by the use of dynamically-configured clients. As such, the use of audience-restricted access tokens and **Resource Indicators [[RFC8707](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-01.html#RFC8707)] is RECOMMENDED** when using the features in this specification.

However there is currently no way for the OAuth client to know (in the general case i.e. loose coupling between the different actors) that the audience restriction has been properly supported and applied by authorization server. For example, the authorization server might silently ignore the `resource` parameter of the authorization request.

Ideally, the authorization server should probably include the `resource` back in the authorization response similar to how this is done for the `scope` parameter (and in a sense for the [`authorization_details` parameter](https://www.ietf.org/archive/id/draft-ietf-oauth-rar-21.html#name-token-response)). Such an option should probably have been included in [RFC8707](https://datatracker.ietf.org/doc/html/rfc8707), however.

Alternatively (or in addition) some authorization server metadata could be used to indicate that it supports resource indication.

**Comment by aaronpk** [comment]
I agree that the token response should include `resource` like it does `scope` to inform the client. However I don't think that is a mechanism that this spec can suggest or define, it probably should have been part of RFC8707. Same goes for the authorization server metadata to indicate support, that should have been part of 8707, and isn't something this spec can define.

**Comment by selfissued** [comment]
I agree with Aaron's assessment that this is out of scope for this specification.  I'm prone to close this issue on that basis.

**Comment by aaronpk** [comment]
This came up in the context of MCP and some discussion has been happening:

* https://github.com/modelcontextprotocol/modelcontextprotocol/issues/763
* https://github.com/oauth-wg/oauth-v2-1/issues/215

There is a new draft that defines the `resource` parameter in the token response:

https://datatracker.ietf.org/doc/draft-mcguinness-oauth-resource-token-resp/

Given that this draft is now RFC9728 and the work to define the `resource` parameter in the token response has started, I'm closing this issue.
